src/Cake.Common/Tools/SpecFlow/TestExecutionReport/SpecFlowTestExecutionReporter.cs

```diff
@@ -0,0 +1,220 @@
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="globber">The globber.</param>
+        public SpecFlowTestExecutionReporter(
+            IFileSystem fileSystem,
+            ICakeEnvironment environment,
+            IProcessRunner processRunner,
+            IGlobber globber)
+            : base(fileSystem, environment, processRunner, globber)
+        {
+            _environment = environment;
+        }
+
+        /// <summary>
+        /// Runs SpecFlow Test Execution Report with the specified settings.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="action">The action.</param>
+        /// <param name="projectFile">The project file path.</param>
+        /// <param name="settings">The settings.</param>
+        public void Run(ICakeContext context,
+            Action<ICakeContext> action,
+            FilePath projectFile,
+            SpecFlowTestExecutionReportSettings settings)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+            if (action == null)
+            {
+                throw new ArgumentNullException("action");
+            }
+            if (projectFile == null)
+            {
+                throw new ArgumentNullException("projectFile");
+            }
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            // Run the tool using the interceptor.
+            var interceptor = InterceptAction(context, action);
+
+            // Get / Verify Arguments
+            var builder = GetArguments(interceptor, settings, projectFile);
+
+            // Execute the action
+            try
+            {
+                action(context);
+            }
+            catch (CakeException e)
+            {
+                // Write warning to log
+                context.Warning(e.Message);
+            }
+
+            // Run the tool.
+            Run(settings, builder);
+        }
+
+        private static SpecFlowContext InterceptAction(
+            ICakeContext context,
+            Action<ICakeContext> action)
+        {
+            var interceptor = new SpecFlowContext(context);
+
+            action(interceptor);
+
+            // Validate arguments.
+            if (interceptor.FilePath == null)
+            {
+                throw new CakeException("No tool was started.");
+            }
+
+            return interceptor;
+        }
+
+        private ProcessArgumentBuilder GetArguments(
+            SpecFlowContext context,
+            SpecFlowTestExecutionReportSettings settings,
+            FilePath projectFile)
+        {
+            var builder = new ProcessArgumentBuilder();
+
+            var executable = context.FilePath.ToString();
+
+            if (executable.IndexOf("mstest", StringComparison.OrdinalIgnoreCase) >= 0)
```

Comment:
Maybe move the content of the three if-statements to their own methods in a separate static class. Would make it easier to follow the code. What do you think?